### ISSUE TITLE
Ratchet & Clank: ACiT [BCES00511] flickering triangle explosions after #8984

### ISSUE THREAD

**Issue by isJuhn** [RPCS3/rpcs3#9128]
## Quick summary
The game has flickering triangles now when you look in certain directions.

## Details
Like vertex explosions but no vertices are affected. See picture. Also rsx captures are quite broken in this game but the black triangles are still there and "flickering" at 1 fps when replaying. You can see it faintly with blueish triangles when you spawn in the great clock but once you get to Clanks subconscious they become black and a lot more prevalent.

#### 1. Please provide _exact_ build (or commit) information that introduced the regression you're reporting.
#8984 

#### 2. Please attach TWO logs:
[Working log](https://github.com/RPCS3/rpcs3/files/5434822/RPCS3.10950.zip)
[Broken log](https://github.com/RPCS3/rpcs3/files/5434823/RPCS3.10953.zip)

#### 3. If you describe graphical regression, please provide an RSX capture and a RenderDoc capture that demonstrate it.
[RSX capture](https://github.com/RPCS3/rpcs3/files/5434826/BCES00511_20201025142647_capture_2.zip)
RenderDocs are in double digit gigabyte range for this game so I will not be posting one of those.

#### 4. Please attach screenshots of your problem.
![yINoSVhEF2](https://user-images.githubusercontent.com/26828095/97108772-e58a0080-16cf-11eb-823e-72660d290c7e.gif)

#### 5. Please provide comparison with real PS3.
I think the flickering speaks for itself.

#### 6. Please provide your system configuration:
Windows 10 2004
Ryzen 7 3700X
Nvidia GTX 1080 ti
Driver 451.67.0.0

#### Please include.
Use save file from #8944. [Here](https://www.youtube.com/watch?v=ZLtbsUmB4IY)'s a walkthrough on how to get to Clanks subconscious.

**Comment by AniLeo** [comment]
> Driver 451.67.0.0

Outdated NVIDIA driver issue, not an RPCS3 bug. Update your drivers.

**Comment by isJuhn** [comment]
Retested on newest driver, newest version, cleaned shader cache, still an issue.

**Comment by AniLeo** [comment]
Gonna test on all vendors to make sure this is not a new issue, reopening in the meantime

**Comment by AniLeo** [comment]
**Mesa RADV 20.2.1, R9 280X**
No issues

![image](https://user-images.githubusercontent.com/10283761/97112272-4408ac00-16db-11eb-8d75-ba14020cfc0e.png)

---

**NVIDIA Windows 456.71, GTX 960M**
Driver crashes when going in-game with the following validation error:

`
E {RSX [0x1693e4c]} RSX: ERROR: [Validation] Code 0 : Validation Error: [ VUID-VkGraphicsPipelineCreateInfo-blendEnable-02023 ] Object 0: handle = 0x27323828, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x1991022a | vkCreateGraphicsPipelines(): pipeline.pColorBlendState.pAttachments[0].blendEnable is VK_TRUE but format VkImage 0x6280a200000018cf[] associated with this attached image (VK_FORMAT_B8G8R8A8_UNORM) does not support VK_FORMAT_FEATURE_COLOR_ATTACHMENT_BLEND_BIT. The Vulkan spec states: If rasterization is not disabled and the subpass uses color attachments, then for each color attachment in the subpass the blendEnable member of the corresponding element of the pAttachment member of pColorBlendState must be VK_FALSE if the attached image's format features does not contain VK_FORMAT_FEATURE_COLOR_ATTACHMENT_BLEND_BIT 
`
(https://vulkan.lunarg.com/doc/view/1.2.148.0/windows/1.2-extensions/vkspec.html#VUID-VkGraphicsPipelineCreateInfo-blendEnable-02023)

**Comment by AniLeo** [comment]
This is an on-going driver bug

I've tested on Beta Driver 456.80 and it still happens
https://developer.nvidia.com/vulkan-beta-45680-windows-78
https://developer.nvidia.com/vulkan-beta-45680-windows-10-dch
 
```
October 9th, 2020 - Windows 456.80, Linux 455.26.01

    Fixes:
        Reduce host memory consumption for descriptor memory when VkDescriptorSetVariableDescriptorCountAllocateInfo is used
        Handle SPIR-V 1.4 non-Input/Ouput entry point variables correctly
        Fixed a blending optimization that sometimes produced an incorrect result
        Fixed SPIR-V intersection shader compilation issue when multi entry point ray tracing modules are used
```

Can't test on 457.00 because the installer on the NVIDIA website is not working here
https://developer.nvidia.com/vulkan-beta-45700-windows-78
https://developer.nvidia.com/vulkan-beta-45700-windows-78-dch

I can't capture a RenderDoc as I run out of memory while doing so even with 16G RAM + 16G Pagefile

**Comment by kd-11** [comment]
Yet another SPIRV driver bug. We'll see about another workaround for this.

**Comment by isJuhn** [comment]
Managed to capture a triangle in a not huge renderdoc, watch out because the renderdoc is cursed though. Sometimes you can see it, then if you look at another event and go back it's suddenly gone...
[banani.rdc](https://drive.google.com/file/d/1IZ6mAYjR20jq0fFxNK0Jb3r3MxeCn-AK/view?usp=sharing)
Here's a screenshot of eid 6787, but it can be seen on a lot of other eids too, just need to be lucky. It's not a black triangle unfortunately as that area of the game will eat through my 32GB of ram like a [hungry kot](https://www.youtube.com/watch?v=2NrGZ_wNIqA).
![image](https://user-images.githubusercontent.com/26828095/97115413-dff3e100-16f6-11eb-8c05-d98783e9c9df.png)

**Comment by kd-11** [comment]
An unrelated regression is keeping this game from booting. Investigation will resume after that is sorted out. In the meantime, if there are other games affected it would be nice to take a look.

**Comment by jeremybris** [comment]
> In the meantime, if there are other games affected it would be nice to take a look.

Resistance 3, also effected, (for me,) @kd-11 .

**Comment by kd-11** [comment]
Fixed by https://github.com/RPCS3/rpcs3/pull/9726. This bug had nothing to do with the PR in question, I'm not sure why exactly the issue only appeared after that PR specifically. Its likely just some other coincidence where the old code generated some unexpected instructions that hid the issue.
